src/display/expanded_display.rs

```diff
@@ -0,0 +1,169 @@
+use crate::Tabled;
+
+/// ExpandedDisplay display data in a 'expanded display mode' from postgress.
+/// It may be usefull for a large data sets with a lot of fields.
+///
+/// See 'Examples' in https://www.postgresql.org/docs/current/app-psql.html.
+///
+/// It escapes strings to resolve a multi-line ones.
+/// Because of that `colors` may not be rendered.
+pub struct ExpandedDisplay {
+    format_record_splitter: Option<fn(usize) -> String>,
+    format_value: Option<Box<dyn Fn(&str) -> String>>,
+    fields: Vec<String>,
+    records: Vec<Vec<String>>,
+}
+
+impl ExpandedDisplay {
+    /// Creates a new instance of ExpandedDisplay
+    pub fn new<T: Tabled>(iter: impl IntoIterator<Item = T>) -> Self {
+        let data = iter.into_iter().map(|i| i.fields()).collect();
+        let header = T::headers();
+
+        Self {
+            records: data,
+            fields: header,
+            format_record_splitter: None,
+            format_value: None,
+        }
+    }
+
+    /// Sets a line format which will be used to split records.
+    ///
+    /// Default formating is "-[ RECORD {} ]-"
+    ///
+    /// At least one '\n' char will be printed at the end regardless if you set it or not.
+    pub fn format_record_head(&mut self, f: fn(usize) -> String) -> &mut Self {
+        self.format_record_splitter = Some(f);
+        self
+    }
+
+    /// Use a value formatter.
+    pub fn format_value(&mut self, f: impl Fn(&str) -> String + 'static) -> &mut Self {
+        self.format_value = Some(Box::new(f));
+        self
+    }
+
+    /// Turn off a wrapping of multiline value.
+    pub fn format_value_in_one_line(&mut self) -> &mut Self {
+        self.format_value = Some(Box::new(|s| s.escape_debug().to_string()));
+        self
+    }
+
+    /// Sets max width of value.
+    /// The rest will be trunceted.
+    pub fn format_value_max_width(&mut self, max: usize) -> &mut Self {
+        self.format_value = Some(Box::new(move |s| {
+            s.chars()
+                .take(max)
+                .collect::<String>()
+                .escape_debug()
+                .to_string()
+        }));
+        self
+    }
+
+    /// Sets max width of value,
+    /// when limit is reached next chars will be placed on the next line.
+    pub fn format_value_max_width_wrapped(&mut self, max: usize) -> &mut Self {
```

Comment:
Could just be called `max_width_wrapped`? Although I agree, I don't really love the naming duplication here, it feels like it would be nice to have methods like:

```rust
impl ExpandedDisplay {
    pub fn max_width(&self, max: usize) -> &mut Self;
    pub fn truncate(&self, truncate: bool) -> &mut Self;
    pub fn wrap(&self, wrap: bool) -> &mut Self;
}
```
where calling `x.wrap(true)` would switch off truncation, and vice versa. But I think that would require generating the `self.format_value` attribute at `fmt`-time, rather than setting it as a boxed closure 🤔

Reply:
what about just having this?

```
impl ExpandedDisplay {
    pub fn truncate(&self, max: usize) -> &mut Self;
    pub fn wrap(&self, max: usize) -> &mut Self;
}
```

Reply:
I renamed it as I propoused

Reply:
Nice, looks good!